<!-- -*- mode: XML -*- -->

<launch>
    <node name="relay_arm_sensor_interface_kit" pkg="phidgets_ros" type="interface_kit.py" output="screen">
        <param name="serial_number" type="int" value="124427"/>
    </node>

    <node name="top_interface_kit" pkg="phidgets_ros" type="interface_kit.py" output="screen">
        <param name="serial_number" type="int" value="106950"/>
    </node>

    <!-- Start the Dynamixel servo manager to control Lower Arm (wrist and fingers) -->
    <node name="lower_arm_motor_manager" pkg="ax12_controller_core" type="controller_manager.py" output="screen">
        <param name="port_name" type="str" value="/dev/wubble2LowerArm"/>
        <param name="baud_rate" type="int" value="1000000"/>
        <param name="max_motor_id" type="int" value="25"/>
        <param name="update_rate" type="int" value="50"/>
    </node>

    <!-- Start all wubble arm joint controllers -->
    <rosparam file="$(find wubble2_robot)/config/wubble_gripper.yaml" command="load"/>
    <node name="wubble_lower_arm_controller_spawner" pkg="ax12_controller_core" type="controller_spawner.py"
          args="--port /dev/wubble2LowerArm
                gripper_left_finger_controller
                gripper_right_finger_controller"
          output="screen"/>
</launch>
